### ISSUE TITLE
Netedit: Edit Traffic Light mode, enhancement at program saving into separate file

### ISSUE THREAD

**Issue by ZoltanBaksa** [eclipse/sumo#7964]
Hi,
In my city simulation I have approx 80 TL controlled intersections.
In most cases there are multiple programs for the same TL controller. I have to save all programs independently from the network, because of customer security reasons I'm not allowed to store it together with the network. I have to select all programs one-by-one and use the 'Save TLS Program' button, creating as many separate files as many programs there are. If I'm done with the editing inside netedit, I have to delete them to meet the security requirement, because when I save the network the programs are also stored.
To use them during simulation, I have to merge these into loadable additionals. To be able to edit them in netedit, I have to load each. If would be helpful, if there would be a button to save all programs into one separate file together and handle them similarly as in case of the additionals.

**Comment by namdre** [comment]
you can use the functions 'save programs' and 'load programs' from the main menu bar: file->'traffic lights'

**Comment by ZoltanBaksa** [comment]
Hi Jakob,
I used the function, but I still need to delete the additionally created programs from the network after I saved them in a separate file.

**Comment by ZoltanBaksa** [comment]
I will try it out tomorrow

**Comment by namdre** [comment]
If you are only editing the tls programs, then you don't even need to save the network so there is nothing you need to delete afterwards.

**Comment by ZoltanBaksa** [comment]
There is an issue with the traffic light load and save process. Here are the steps I followed:
- The original network contains only TL programs with id 0, containing off-blinking state.
- I load the additional programs stored in a separate file.
- Change a program part for one TL controller
- Save the TL programs again in another file to see the differences
The new file contains only the original programs and the one that was changed. All other loaded programs are not stored in the new file.
If I check other TL controlled intersections by clicking them in TL mode, the stored additional file contains their program to.
